updater/library/src/updater.rs

```diff
@@ -172,7 +172,7 @@ fn get_base_path(original_lib_app_paths: &Vec<String>) -> anyhow::Result<PathBuf
     // Iterate through the paths and find the first one that exists.
     for path in original_lib_app_paths {
         let path = PathBuf::from(path);
-        if path.exists() {
+        if path.try_exists()? {
```

Comment:
This changes the behavior of the method so that if there is an earlier path with permissions errors we will now error out of get_base_path, rather than previously we would have returned a valid later path.

Is that what you want?

Reply:
https://doc.rust-lang.org/std/fs/fn.try_exists.html

Reply:
yeah good point I'll adjust the behavior